src/Features/CSharp/Portable/ExtractInterface/CSharpExtractInterfaceService.cs

```diff
@@ -30,25 +31,24 @@ internal sealed class CSharpExtractInterfaceService : AbstractExtractInterfaceSe
         public CSharpExtractInterfaceService()
         {
         }
-
         protected override async Task<SyntaxNode> GetTypeDeclarationAsync(Document document, int position, TypeDiscoveryRule typeDiscoveryRule, CancellationToken cancellationToken)
         {
-            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-            var root = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
-            var token = root.FindToken(position != tree.Length ? position : Math.Max(0, position - 1));
-            var typeDeclaration = token.GetAncestor<TypeDeclarationSyntax>();
+            var span = new TextSpan(position, 0);
+            var nodes = await document.GetRelevantNodesAsync<SyntaxNode>(span, cancellationToken).ConfigureAwait(false);
+
+            var typeDeclaration = nodes.Where(n => n is TypeDeclarationSyntax).FirstOrDefault() ?? nodes.FirstOrDefault();
```

Comment:
Maybe use [`document.TryGetRelevantNodeAsync<TypeDeclarationSyntax>(span, cancellationToken)`](https://github.com/dotnet/roslyn/blob/main/src/Features/Core/Portable/CodeRefactorings/CodeRefactoringContextExtensions.cs#L53) instead? So we don't need some of those checks, etc.

Reply:
This is a great suggestion. I ran into one problem with TryGetRelevantNode. For the case of triggering the action from a closing brace, TryGetRelevantNode returns only the CompilationUnitSyntax. GetRelevantNodes returns both the CompilationUnitSyntax and the ClassDeclarationSyntax. I'm having trouble with how to approach that case. 
```
class MyClass
{
  public void MyMethod()
  {

  }
}$$
```

Reply:
That feels like an issue in `TryGetRelevantNode`. At the minimum, I'd expect `TryGetRelevantNode<TypeDeclarationSyntax>` and `GetRelevantNodes<TypeDeclarationSyntax>` would return the same list of nodes. I'd suggest to fix `TryGetRelevantNode` and see if anything is broken :)

Reply:
> For the case of triggering the action from a closing brace ...

I don't think we need it to work from that location.

Reply:
> I don't think we need it to work from that location.

That works for me too. My concern is mainly about the inconsistency between those two methods.

Reply:
Grabbing the TypeDeclarationSyntax first solved this problem. The closing brace location works again.